src/server/services/wms/qgswmsparameters.h

```diff
@@ -370,6 +372,50 @@ namespace QgsWms
        */
       int heightAsInt() const;
 
+      /**
+       * Returns SRCWIDTH parameter or an empty string if not defined.
+       * \returns srcWidth parameter
+       */
+      QString srcWidth() const;
+
+      /**
+       * Returns SRCWIDTH parameter as an int or its default value if not
+       * defined. An exception is raised if SRCWIDTH is defined and cannot be
+       * converted.
+       * \returns srcWidth parameter
+       * \throws QgsBadRequestException
+       */
+      int srcWidthAsInt() const;
+
+      /**
+       * Returns SRCHEIGHT parameter or an empty string if not defined.
+       * \returns srcHeight parameter
+       */
+      QString srcHeight() const;
+
+      /**
+       * Returns SRCHEIGHT parameter as an int or its default value if not
+       * defined. An exception is raised if SRCHEIGHT is defined and cannot be
+       * converted.
+       * \returns srcHeight parameter
+       * \throws QgsBadRequestException
+       */
+      int srcHeightAsInt() const;
+
+      /**
+       * Returns SRCHEIGHT parameter if it's a GetLegendGraphics request and otherwise HEIGHT parameter
+       *
+       * \returns getHeightAsInt parameter
+       */
+      int getHeightAsInt() const;
```

Comment:
Can you come up with a more descriptive name here? The API looks quite confusing when reading the changes below -- ("heightAsInt()" -> "getHeightAsInt()" ) ... looks very likely to be misused.